### PR TITLE
[Bugfix] timing_hash.rb:8:in `each': comparison of Hash with 1438133594940 failed (ArgumentError)

### PR DESCRIPTION
**Pull request by toshiyuki-hasegawa-TL**
Could you please check the implementation and working?
# Summary
- This PR fixed the this error
  -  `timing_hash.rb:8:in each: comparison of Hash with 1438133594940 failed (ArgumentError)`.
- This is a fix for this problem. 
  - https://github.com/desktoppr/wbench/issues/31#issuecomment-125814114 
# Fixes
- Maybe, I think window.performance.timing response was modified.
  So fixed to calculate the correct result from the modified response.
- Rspec is not working, so change RSpec 3 syntax in Transpec.
  - Transpec : https://github.com/yujinakayama/transpec
# error

I receive this error since July 23nd, 2015 (JP time).
It always error occurs.
It did not receive an error until July 22, 2015 (JP time).

Mac : OS X Yosemite 10.10.4
Ruby : ruby 2.1.3p242 (2014-09-19 revision 47630) [x86_64-darwin14.0]
Chrome : 44.0.2403.107 (64-bit)

``` bash
$ ./bin/wbench -l 1 http://www.google.com
/Users/toshiyuki_hasegawa/workspace/wbench/lib/wbench/timing_hash.rb:8:in `each': comparison of Hash with 1438133594940 failed (ArgumentError)
        from /Users/toshiyuki_hasegawa/workspace/wbench/lib/wbench/timing_hash.rb:8:in `min_by'
        from /Users/toshiyuki_hasegawa/workspace/wbench/lib/wbench/timing_hash.rb:8:in `initialize'
        from /Users/toshiyuki_hasegawa/workspace/wbench/lib/wbench/timings/browser.rb:9:in `new'
        from /Users/toshiyuki_hasegawa/workspace/wbench/lib/wbench/timings/browser.rb:9:in `result'
        from /Users/toshiyuki_hasegawa/workspace/wbench/lib/wbench/benchmark.rb:32:in `browser_results'
        from /Users/toshiyuki_hasegawa/workspace/wbench/lib/wbench/benchmark.rb:20:in `block (3 levels) in run'
        from /Users/toshiyuki_hasegawa/workspace/wbench/lib/wbench/browser.rb:34:in `visit'
        from /Users/toshiyuki_hasegawa/workspace/wbench/lib/wbench/benchmark.rb:20:in `block (2 levels) in run'
        from /Users/toshiyuki_hasegawa/workspace/wbench/lib/wbench/benchmark.rb:18:in `times'
        from /Users/toshiyuki_hasegawa/workspace/wbench/lib/wbench/benchmark.rb:18:in `block in run'
        from /Users/toshiyuki_hasegawa/workspace/wbench/lib/wbench/benchmark.rb:17:in `tap'
        from /Users/toshiyuki_hasegawa/workspace/wbench/lib/wbench/benchmark.rb:17:in `run'
        from /Users/toshiyuki_hasegawa/workspace/wbench/lib/wbench/benchmark.rb:4:in `run'
        from ./bin/wbench:43:in `<main>'
```

### CODE REVIEW

File: lib/javascripts/wbench.js [hunk 1]
```diff
@@ -99,11 +99,18 @@ window.WBench = {
 
     // Use user defined timings that are added via window.performance.mark('event name')
     if(typeof(window.performance.getEntriesByType) === 'function') {
+      results = {};
+
+      for (var key in resultTimings) {
+        if( !resultTimings[key] || key === 'toJSON' ) continue;
```
Comment (mariovisic) [inline lib/javascripts/wbench.js]: Thanks for the patch, I really appreciate it.

It seems like the issue is caused by `window.performance.timing` not returning a JSON object but another type of object that doesn't serialise correctly. Can you explain to me why this line is required? I don't fully understand. It seems like we could fix the problem by changing line 98 from being:

```ruby
resultTimings = window.performance.timing;
```
to
```ruby
resultTimings = window.performance.timing.toJSON();
```

Comment (toshiyuki-hasegawa-TL) [inline lib/javascripts/wbench.js]: Thank you for checking and response.

You are right! I was a misunderstanding.
I thought has changed in unix time for response of `window.performance.timing`.
But it was a mistake.

So it was fixed.

Could you please check ?

**Comment by mariovisic** [comment]
Yup, thanks for that change it looks much better. Does this work in firefox as well? I'm not sure if other browsers also return an object that responds to `#toJSON`. Do we need to add a check?

**Comment by toshiyuki-hasegawa-TL** [comment]
Yes, It seems work if firefox even by applying this fix.
Because it works for my local machine.

Mac : OS X Yosemite 10.10.4
Ruby : ruby 2.2.2p95 (2015-04-13 revision 50295) [x86_64-darwin14]
Firefox : 39.0.3

I checked firefox for master branch but has occurred same like error.
```
## master
./bin/wbench -l 1 -b firefox http://google.com
/Users/hasegawa/workspace/wbench/lib/wbench/timing_hash.rb:8:in `each': comparison of Fixnum with String failed (ArgumentError)
        from /Users/hasegawa/workspace/wbench/lib/wbench/timing_hash.rb:8:in `min_by'
        from /Users/hasegawa/workspace/wbench/lib/wbench/timing_hash.rb:8:in `initialize'
        from /Users/hasegawa/workspace/wbench/lib/wbench/timings/browser.rb:9:in `new'
        from /Users/hasegawa/workspace/wbench/lib/wbench/timings/browser.rb:9:in `result'
        from /Users/hasegawa/workspace/wbench/lib/wbench/benchmark.rb:32:in `browser_results'
        from /Users/hasegawa/workspace/wbench/lib/wbench/benchmark.rb:20:in `block (3 levels) in run'
        from /Users/hasegawa/workspace/wbench/lib/wbench/browser.rb:34:in `visit'
        from /Users/hasegawa/workspace/wbench/lib/wbench/benchmark.rb:20:in `block (2 levels) in run'
        from /Users/hasegawa/workspace/wbench/lib/wbench/benchmark.rb:18:in `times'
        from /Users/hasegawa/workspace/wbench/lib/wbench/benchmark.rb:18:in `block in run'
        from /Users/hasegawa/workspace/wbench/lib/wbench/benchmark.rb:17:in `tap'
        from /Users/hasegawa/workspace/wbench/lib/wbench/benchmark.rb:17:in `run'
        from /Users/hasegawa/workspace/wbench/lib/wbench/benchmark.rb:4:in `run'
        from ./bin/wbench:43:in `<main>'
```

**Comment by mariovisic** [comment]
Thanks for your contribution, i'll push a new gem release shortly :)

**Pull request merged**